zipkin-cassandra-core/src/main/java/org/twitter/zipkin/storage/cassandra/Repository.java

```diff
@@ -70,6 +70,23 @@
     private final PreparedStatement selectTraceDurationTail;
     private final Map<String,String> metadata;
 
+    private final ThreadLocal<Set<String>> writtenNames = new ThreadLocal<Set<String>>() {
+                private long hour = TimeUnit.MILLISECONDS.toHours(System.currentTimeMillis());
```

Comment:
Limiting the size sounds good, I think evicting would be important to prevent spill over and cache would always be warm (maybe thread local lrumap O_o).  Sounds like we could just revisit these issues should they surface.